R/dashboard.R

```diff
@@ -0,0 +1,51 @@
+#' R Markdown format for model monitoring dashboards
+#'
+#' @inheritParams pins::pin_read
+#' @param ... Arguments passed to [flexdashboard::flex_dashboard()]
+#' @param display_pins Should the dashboard display a link to the pin(s)?
+#' Defaults to `TRUE`.
+#' @param display_api Should the dashboard display the OpenAPI documentation
+#' for the deployed model? Defaults to `TRUE`.
+#'
+#' @details The `vetiver_dashboard()` function is a specialized type of
+#' \pkg{flexdashboard}. See the flexdashboard website for additional
+#' documentation:
+#'  \href{http://rmarkdown.rstudio.com/flexdashboard/}{http://rmarkdown.rstudio.com/flexdashboard/}
+#'
+#' @export
+vetiver_dashboard <- function(board, name, version = NULL,
+                              ...,
+                              display_pins = TRUE, display_api = TRUE) {
+
+    dashboard_dots <- rlang::list2(...)
+
+    if (board$versioned) {
+        if (is_null(version)) {
+            version <- pins::pin_versions(board, name)
+            version <- choose_version(version)
+        }
+        v <- vetiver_pin_read(board, name, version = version)
+    } else {
+        v <- vetiver_pin_read(board, name)
+    }
+
+    if(rlang::has_name(dashboard_dots, "navbar")) {
+        navbar <- dashboard_dots$navbar
+    } else {
+        navbar <- NULL
+    }
+    dashboard_dots <- modifyList(dashboard_dots, list(navbar = NULL))
+
+    if (display_pins) {
+        pin_navbar <- list(
+            title = "Model Pin",
+            icon = "fa-location-dot",
+            href = v$metadata$url,
+            target = "_blank"
+        )
+        navbar <- append(navbar, list(pin_navbar))
+    }
+
+    flexdashboard::flex_dashboard(!!!dashboard_dots, navbar = navbar)
```

Comment:
You probably need to use `rlang::inject()` here (https://rlang.r-lib.org/reference/inject.html) because you are using `!!!` operator in a function that does not have built-in support for it. 

**rlang** documents this as out of context usage https://rlang.r-lib.org/reference/topic-inject-out-of-context.html#using-and-out-of-context

```suggestion
    rlang::inject(flexdashboard::flex_dashboard(!!!dashboard_dots, navbar = navbar))
```